receiver/receivercreator/observerhandler.go

```diff
@@ -19,21 +19,25 @@ import (
 	"sync"
 
 	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/consumer"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
 )
 
-var _ observer.Notify = (*observerHandler)(nil)
+var (
+	_ observer.Notify = (*observerHandler)(nil)
+)
 
 // observerHandler manages endpoint change notifications.
 type observerHandler struct {
 	sync.Mutex
+	config *Config
 	logger *zap.Logger
-	// receiverTemplates maps receiver template full name to a receiverTemplate value.
-	receiverTemplates map[string]receiverTemplate
 	// receiversByEndpointID is a map of endpoint IDs to a receiver instance.
 	receiversByEndpointID receiverMap
+	// nextConsumer is the receiver_creator's own consumer
+	nextConsumer consumer.MetricsConsumer
```

Comment:
Looks like we are more and more specializing the receiver creator to be usable only by metric receivers. Was this the initial intent? It does not appear that observer itself cares about the signal.

Reply:
With how consumers are structured (and perhaps lack of generics in Go) I don't think there's a generic way to handle all types of telemetry since you have the `pdata.{Metrics,Traces,Logs}` types that have resources within them but no generic way to process resources without knowing the discrete telemetry types. I think if we find a use case for another telemetry type (logs perhaps?) I don't think it'd be that hard to add to receivercreator.